worldedit-core/src/main/java/com/fastasyncworldedit/core/util/TextureUtil.java

```diff
@@ -352,17 +358,22 @@ public TextureUtil(File folder) throws FileNotFoundException {
             try {
                 LOGGER.info("Downloading asset jar from Mojang, please wait...");
                 new File(Fawe.platform().getDirectory() + "/" + Settings.settings().PATHS.TEXTURES + "/").mkdirs();
-                try (BufferedInputStream in = new BufferedInputStream(
-                        new URL("https://piston-data.mojang.com/v1/objects/c0898ec7c6a5a2eaa317770203a1554260699994/client.jar")
-                                .openStream());
-                     FileOutputStream fileOutputStream = new FileOutputStream(
-                             Fawe.platform().getDirectory() + "/" + Settings.settings().PATHS.TEXTURES + "/1.19.2.jar")) {
-                    byte[] dataBuffer = new byte[1024];
-                    int bytesRead;
-                    while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
-                        fileOutputStream.write(dataBuffer, 0, bytesRead);
+
+                try {
+                    VersionMetadata metadata = getLatestVersion();
+                    LOGGER.info("Latest release version is {}", metadata.version());
+                    String url = getLatestClientJarUrl(metadata);
+                    try (BufferedInputStream in = new BufferedInputStream(new URL(url).openStream());
+                         FileOutputStream fileOutputStream = new FileOutputStream(
+                                 Fawe.platform().getDirectory() + "/" + Settings.settings().PATHS.TEXTURES +
+                                         "/" + metadata.version() + ".jar")) {
+                        byte[] dataBuffer = new byte[1024];
+                        int bytesRead;
+                        while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
+                            fileOutputStream.write(dataBuffer, 0, bytesRead);
+                        }
+                        LOGGER.info("Asset jar down has been downloaded successfully.");
```

Comment:
I see you copied that from the line below, but grammar wise, the sentence doesn't make much sense.

I have no idea what I meant by that.